examples/platform/k32w/app/support/Memconfig.cpp

```diff
@@ -0,0 +1,110 @@
+/*
+ *
+ *    Copyright (c) 2020-2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file implements heap memory allocation APIs for CHIP. These functions are platform
+ *      specific and might be C Standard Library heap functions re-direction in most of cases.
+ *
+ */
+
+#include "FreeRTOS.h"
+#include <cstring>
+#include <stdlib.h>
+
+#ifndef NDEBUG
+#include <atomic>
+#include <cstdio>
+#endif
+
+#if CHIP_CONFIG_MEMORY_DEBUG_DMALLOC
+#include <dmalloc.h>
+#include <support/SafeInt.h>
+#endif // CHIP_CONFIG_MEMORY_DEBUG_DMALLOC
+
+namespace chip {
+namespace Platform {
+
+extern "C" {
+
+void * __wrap_malloc(size_t size)
+{
+    return pvPortMalloc(size);
+}
+
+void __wrap_free(void * ptr)
+{
+    vPortFree(ptr);
+}
+
+void * __wrap_calloc(size_t num, size_t size)
+{
+    size_t total_size = num * size;
+
+    if (size && total_size / size != num)
+        return nullptr;
+
+    void * ptr = pvPortMalloc(total_size);
+    if (ptr)
+        memset(ptr, 0, total_size);
+    return ptr;
+}
+
+void * __wrap_realloc(void * ptr, size_t new_size)
+{
+    if (new_size)
+    {
+        void * new_ptr = pvPortMalloc(new_size);
+        if (new_ptr)
+        {
+            if (ptr)
+            {
+                // possibly copying beyond ptr + size
+                memcpy(new_ptr, ptr, new_size);
```

Comment:
Should not copy beyond ptr+size. This code likely came from a previous template, where realloc usage patterns were known. For CHIP, it would be best to validate that you never read more bytes from the source than there were.

Reply:
Implemented xPortMallocUsableSize, which helps me make sure that the newly allocated size is bigger than the older one.